packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts

```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
+	public top3: Athlete[] = [];
+	private _live = true;
+	private _timer: any;
+	private windowWidth: any;
+	private _overlayId: string = '';
+  
+	get live() {
+	  return this._live;
+	}
+  
+	set live(val) {
+	  this._live = val;
+	  if (this._live) {
+		this._timer = setInterval(() => this.ticker(), 1500);
+	  } else {
+		clearInterval(this._timer);
+	  }
+	}
+  
+	get showWinnerOverlay() {
+	  return this.showOverlay && this.hasWinner && !this.isFinished;
+	}
+  
+	get showFinishedOverlay() {
+	  return this.showOverlay && this.isFinished;
+	}
+  
+	get hideAthleteNumber() {
+	  return this.windowWidth && this.windowWidth < 960;
+	}
+	get hideBeatsPerMinute() {
+	  return (this.windowWidth && this.windowWidth < 860) || !this.live;
+	}
+  
+	constructor(@Inject(IgxOverlayService) public overlayService: IgxOverlayService) { }
+	public ngOnInit() {
+	  this.localData = athletesData.slice(0, 30).sort((a, b) => b.TrackProgress - a.TrackProgress);
+	  this.localData.forEach(rec => this.getSpeed(rec));
+	  this.windowWidth = window.innerWidth;
+	  this._timer = setInterval(() => this.ticker(), 1500);
+	  this.overlayService.onClosing.subscribe((event: OverlayClosingEventArgs) => {
+		this.showOverlay = false;
+	  });
+	}
+  
+	public ngAfterViewInit() {
+	  this.overlaySettings = IgxOverlayService.createAbsoluteOverlaySettings(
+		AbsolutePosition.Center,
+		this.grid1
+	  );
+	  this.overlaySettings.modal = true;
+	}
+  
+	public getValue(cell: IgxGridCellComponent): number {
+	  const val = cell.value;
+	  return val;
+	}
+	public ngOnDestroy() {
+	  clearInterval(this._timer);
+	}
+  
+	public isTop3(cell: IgxGridCellComponent): boolean {
+	  const top = this.grid1.page === 0 && cell.row.index < 4;
+	  return top;
+	}
+  
+	public getTrophyUrl(index: number) {
+	  if (index === 0) {
+		return 'assets/images/trophy_gold.svg';
+	  }
+	  if (index === 1) {
+		return 'assets/images/trophy_silver.svg';
+	  }
+	  if (index === 2) {
+		return 'assets/images/trophy_bronze.svg';
+	  }
+	}
+  
+	public getIconType(cell: IgxGridCellComponent): string {
+	  switch (cell.row.rowData.Position) {
+		case 'current':
+		  return 'arrow_forward';
+		case 'down':
+		  return 'arrow_downward';
+		case 'up':
+		default:
+		  return 'arrow_upward';
+	  }
+	}
+  
+	public getBadgeType(cell: IgxGridCellComponent): string {
+	  switch (cell.row.rowData.Position) {
+		case 'current':
+		  return 'warning';
+		case 'down':
+		  return 'error';
+		case 'up':
+		default:
+		  return 'success';
+	  }
+	}
+  
+	public getSpeed(athlete: Athlete): void {
+	  athlete['Speed'] = this.addSpeedeData(athlete, 40);
+	}
+  
+	public getSpeedeData(minutes?: number): any[] {
+	  if (minutes === undefined) {
+		minutes = 20;
+	  }
+	  const speed: any[] = [];
+	  for (let m = 0; m < minutes; m += 3) {
+		const value = this.getRandomNumber(17, 20);
+		speed.push({ Speed: value, Minute: m });
+	  }
+	  return speed;
+	}
+  
+	public addSpeedeData(athlete: Athlete, minutes?: number): any[] {
```

Comment:
there is a typo in the method name
```suggestion
	public addSpeedData(athlete: Athlete, minutes?: number): any[] {
```